File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string phone_number) {
    string star;
    for(int i = 0; i < phone_number.size() - 4; i++) {
        star.push_back('*');
    }
    
    return phone_number.replace(0, phone_number.size() - 4, star);
}

File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.java
class Solution {
    public String solution(String phone_number) {
        String answer = "";
        int index = phone_number.length();
        
        answer = phone_number.substring(0, index - 4).replaceAll("[0-9]", "*") + phone_number.substring(index - 4, index); 
        
        return answer;
    }
}